flocker/node/functional/test_docker.py

```diff
@@ -367,6 +367,23 @@ def added(_):
         d.addCallback(added)
         return d
 
+    def test_null_environment(self):
+        """
+        A container that does not include any environment variables contains
+        an empty ``environment`` in the return ``Unit``.
+        """
+        client = self.make_client()
+        name = random_name()
+        self.addCleanup(client.remove, name)
+        d = client.add(name, u"busybox:latest")
+        d.addCallback(lambda _: client.list())
+
+        def got_list(units):
+            unit = [unit for unit in units if unit.name == name][0]
```

Comment:
:question: 

`list(generator expression)` is almost always better than `[list comprehension]` if for no other reason than it doesn't leak a name into the local scope (here the `unit` local gets bound once per loop iteration and then it gets bound again at the end to the actual result).